internal/core/item_tree.rs

```diff
@@ -548,7 +548,9 @@ impl ItemRc {
                 (full_transform, parent.children_transform())
             {
                 full_transform = Some(
-                    transform.then_translate(geometry.origin.to_vector()).then(&children_transform),
+                    transform
+                        .then_translate(geometry.origin.to_vector().map(|v| v as _))
```

Comment:
I think we do something similar in a couple of places and the use Euclid's `.cast()`.

```suggestion
                        .then_translate(geometry.origin.to_vector().cast()
```